### PR TITLE
Mapping to account info

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

File: src/AcceptanceTests/Sending/When_dispatching_to_another_account_with_registered_endpoint.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    public class When_dispatching_to_another_account_with_registered_endpoint : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public Task Account_mapped_should_be_respected()
+        {
+            return RunTest(AnotherAccountName);
+        }
+
+        static async Task RunTest(string connectionStringOrName)
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Sending/When_dispatching_to_another_account_with_registered_endpoint.cs]: Is this static method needed? Any reason not to have the contents of this method in the test method itself?

**Comment by SeanFeldman** [comment]
@danielmarbach I've added a`sqacceptancetesting3` & `asqacceptancetesting4` to the pool of storage accounts for this project in TC and configured `AzureStorageQueueTransport_ConnectionString_2` to use it.

**Comment by SeanFeldman** [comment]
Tests are green except API that needs to be re-approved.

**Comment by yvesgoeleven** [comment]
@SeanFeldman Thanks!

**Comment by yvesgoeleven** [comment]
@Particular/azure-maintainers ready for final review

File: src/AcceptanceTests/Sending/When_dispatching_to_other_account.cs [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
+    public class When_dispatching_to_another_account_using_aliases : NServiceBusAcceptanceTest
     {
         [Test]
         public void Connection_string_should_throw()
         {
-            Assert.ThrowsAsync<KeyNotFoundException>(() => RunTest(MainNamespaceConnectionString));
+            Assert.ThrowsAsync<Exception>(() => RunTest(MainNamespaceConnectionString));
         }
 
         [Test]
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Sending/When_dispatching_to_other_account.cs]: Still don't understand why we have to have a separate static method for this test.

Comment (yvesgoeleven) [reply src/AcceptanceTests/Sending/When_dispatching_to_other_account.cs]: In this case it are 2 test with different input and expectations, I changed it for the other tests though

File: src/Transport/Config/Extensions/AzureStorageAddressingSettings.cs [hunk 3]
```diff
@@ -26,33 +29,44 @@ internal void RegisterMapping(string defaultConnectionStringAlias, Dictionary<st
...
+            if (aliasToAccountInfoMap.TryGetValue(account, out accountInfo) == false)
             {
                 if (useLogicalQueueAddresses == false)
                 {
-                    return new ConnectionString(alias);
+                    return new ConnectionString(account);
                 }
 
-                throw new KeyNotFoundException($"No account was mapped under following name '{alias}'. Please map it using AddStorageAccount method.");
+                throw new Exception($"No account was mapped under following name '{address.StorageAccount}'. Please map it using AddStorageAccount method.");
```
[the remark this answers is not in the document]
Comment (yvesgoeleven) [reply src/Transport/Config/Extensions/AzureStorageAddressingSettings.cs]: That is wrong indeed, will change it

File: src/Transport/Config/Extensions/AccountInfo.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    /// 
+    /// </summary>
+    public class AccountInfo
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        public AccountInfo(string alias, string connectionString)
+        {
+            Guard.AgainstNullAndEmpty(nameof(connectionString), connectionString);
```
[the remark this answers is not in the document]
Comment (yvesgoeleven) [reply src/Transport/Config/Extensions/AccountInfo.cs]: addressed

File: src/Transport/Config/Extensions/AzureStorageAddressingSettings.cs [hunk 5]
```diff
@@ -7,7 +7,10 @@
 
     class AzureStorageAddressingSettings
     {
-        internal void RegisterMapping(string defaultConnectionStringAlias, Dictionary<string, string> aliasToConnectionStringMap, bool shouldUseAccountAliases)
+
+        QueueAddressGenerator addressGenerator;
```
[the remark this answers is not in the document]
Comment (yvesgoeleven) [reply src/Transport/Config/Extensions/AzureStorageAddressingSettings.cs]: nitpicker :p

File: src/Transport/Config/Extensions/AccountInfo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (yvesgoeleven) [reply src/Transport/Config/Extensions/AccountInfo.cs]: going to revert this, looks like empty alias is totally expected. There is even a check somewhere "Don't use default empty name for mapping connection strings" implyng empty is supposed to be the default

[the remark this answers is not in the document]
Comment (yvesgoeleven) [reply src/Transport/Config/Extensions/AccountInfo.cs]: Trying null guard only now, that should work

File: src/Transport/Config/Extensions/AzureStorageAddressingSettings.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (SeanFeldman) [reply src/Transport/Config/Extensions/AzureStorageAddressingSettings.cs]: 😃

File: src/Transport/Config/Extensions/AccountInfo.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using AzureStorageQueues;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public class AccountInfo
```
Comment (danielmarbach) [inline src/Transport/Config/Extensions/AccountInfo.cs]: @yvesgoeleven I'll add some doco to this class when I merge the release branch into develop

**Pull request merged**